src/jit/lsra.cpp

```diff
@@ -720,10 +720,10 @@ void LinearScan::applyCalleeSaveHeuristics(RefPosition* rp)
 
     Interval* theInterval = rp->getInterval();
 
-#ifdef DEBUG
-    regMaskTP calleeSaveMask = calleeSaveRegs(getRegisterType(theInterval, rp));
+#if defined(DEBUG)
     if (doReverseCallerCallee())
     {
+        regMaskTP calleeSaveMask = calleeSaveRegs(theInterval->registerType);
```

Comment:
Relocation of this line inside the `if (doReverseCallerCallee())` was my original attempt to workaround the issue.  Since the `if` is only true in special debug code, it disabled the assertion.

Later analysis suggested the switch from `getRegisterType(theInterval, rp)` to `theInterval->registerType` to eliminate the assertion which seemed bogus.

I chose to leave the code relocation, because I prefer to declare variables in the smallest possible scope.  So `calleeSaveMask` should be declalred inside the block.

Reply:
I'm fine with that change, but the `#if defined(DEBUG)` should be removed for consistency (we usually use the other form).